url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ThreadStarter.java

```diff
@@ -0,0 +1,311 @@
+		sb.append("Gloves:\n");
+		sb.append(glov);
+		sb.append("\n\n");
+		sb.append("Armors:\n");
+		sb.append(armors);
+		sb.append("\n\n");
+		sb.append("Helms:\n");
+		sb.append(helms);
+		for (Shoes s : shoes) {
+			s.getItemset().calcBonus();
+			setnames.add(s.getItemset().getName());
+			sets.add(s.getItemset());
+			s.calcBonus();
+		}
+		for (Pants s : pants) {
+			s.getItemset().calcBonus();
+			setnames.add(s.getItemset().getName());
+			sets.add(s.getItemset());
+			s.calcBonus();
+		}
+		for (Gloves s : glov) {
+			s.getItemset().calcBonus();
+			setnames.add(s.getItemset().getName());
+			sets.add(s.getItemset());
+			s.calcBonus();
+		}
+		for (Armor s : armors) {
+			s.getItemset().calcBonus();
+			setnames.add(s.getItemset().getName());
+			sets.add(s.getItemset());
+			s.calcBonus();
+		}
+		for (Helm s : helms) {
+			s.getItemset().calcBonus();
+			setnames.add(s.getItemset().getName());
+			sets.add(s.getItemset());
+			s.calcBonus();
+		}
+		sb.append("");
+		sb.append("Sets:");
+		sb.append(setnames);
+
+		int[] maxis = calcBestSets(sets);
+
+		System.out.println("\nSets:\n" + setnames);
+
+		String i_anzahl = "sets " + setnames.size() + "\tshoes " + shoes.size()
+				+ "\tpants " + pants.size() + "\tgloves " + glov.size()
+				+ "\tarmors " + armors.size() + "\thelms " + helms.size()
+				+ "\n";
+		String i_res = "Results: " + results + "\t"
+				+ defc.getClass().getSimpleName().replace("Comparator", "")
+				+ "\tPrice " + price + "\n";
+		String i_atk = "Atk: " + atk + "\tMatk: " + matk + "\n";
+		String i_comp = df.format(total) + " Comparisons." + "\tCPUs: " + cpus
+				+ "\tTasks:" + shoes.size() + "\n";
+		System.out.println(i_anzahl + i_res + i_atk + i_comp);
+		System.out
+				.println("Each dot is 1m checks. x means 1 Task finished. 1 Task has "
+						+ df.format(total / shoes.size()) + " comparisons.");
+
+		long start = System.currentTimeMillis();
+		try {
+			// 1/4 Loadbalancing - Forking
+			List<Callable<List<Combo>>> tasks = new ArrayList<>();
+			for (int i = 0; i < shoes.size(); i++) {
+				Callable<List<Combo>> c = new ComboCall(price, atk, matk,
+						maxis, shoes.subList(i, i + 1), pants, glov, armors,
+						helms, results, defc);
+				tasks.add(c);
+			}
+
+			// 2/4 Running
+			ExecutorService executor = Executors.newFixedThreadPool(cpus);
+			List<Future<List<Combo>>> f_combos = executor.invokeAll(tasks);
+
+			// 3/4 Joining Results
+			for (Future<List<Combo>> res : f_combos) {
+				combs.addAll(res.get());
+				if (combs.size() > results * 2) {
+					Collections.sort(combs, defc);
+					combs = new ArrayList<>(combs.subList(0, results));
+				}
+			}
+
+			Collections.sort(combs, defc);
+
+			if (results > total) {
+				results = (int) total;
+			}
+
+			if (results > combs.size()) {
+				results = combs.size();
+			}
+
+			combs = new ArrayList<Combo>(combs.subList(0, results));
+
+			// 4/4 Closing
+			executor.shutdown();
+		} catch (InterruptedException | ExecutionException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)